dns_server.go

```diff
@@ -0,0 +1,106 @@
+	Groups                []GroupLabel   `json:"groups,omitempty"`
+	LiveShare             int            `json:"liveShare,omitempty"`
+	ModifiedBy            string         `json:"modifiedBy,omitempty"`
+	ModifiedDate          string         `json:"modifiedDate,omitempty"`
+	SavedEvent            int            `json:"savedEvent,omitempty"`
+	SharedWithAccounts    []AccountGroup `json:"sharedWithAccounts,omitempty"`
+	TestId                int            `json:"testId,omitempty"`
+	TestName              string         `json:"testName,omitempty"`
+	Type                  string         `json:"type,omitempty"`
+	BandwidthMeasurements int            `json:"bandwidthMeasurements,omitempty"`
+	BgpMeasurements       int            `json:"bgpMeasurements,omitempty"`
+	BgpMonitors           []Monitor      `json:"bgpMonitors,omitempty"`
+	DNSServers            []Server       `json:"dnsServers,omitempty"`
+	DNSTransportProtocol  string         `json:"dnsTransportProtocol,omitempty"`
+	Domain                string         `json:"domain,omitempty"`
+	Interval              int            `json:"interval,omitempty"`
+	MtuMeasurements       int            `json:"mtuMeasurements,omitempty"`
+	NetworkMeasurements   int            `json:"networkMeasurements,omitempty"`
+	NumPathTraces         int            `json:"numPathTraces,omitempty"`
+	ProbeMode             string         `json:"probeMode,omitempty"`
+	Protocol              string         `json:"protocol,omitempty"`
+	RecursiveQueries      int            `json:"recursiveQueries,omitempty"`
+}
+
+// AddAgent - Add dns server test
+func (t *DNSServer) AddAgent(id int) {
+	agent := Agent{AgentId: id}
+	t.Agents = append(t.Agents, agent)
+}
+
+// GetDNSServer - get dns server test
+func (c *Client) GetDNSServer(id int) (*DNSServer, error) {
+	resp, err := c.get(fmt.Sprintf("/tests/%d", id))
+	if err != nil {
+		return &DNSServer{}, err
+	}
+	var target map[string][]DNSServer
+	if dErr := c.decodeJSON(resp, &target); dErr != nil {
+		return nil, fmt.Errorf("Could not decode JSON response: %v", dErr)
+	}
+	return &target["test"][0], nil
+}
+
+// CreateDNSServer - Create dns server test
+func (c Client) CreateDNSServer(t DNSServer) (*DNSServer, error) {
+	resp, err := c.post("/tests/dns-server/new", t, nil)
+	if err != nil {
+		return &t, err
+	}
+	if resp.StatusCode != 201 {
+		return &t, fmt.Errorf("failed to create test, response code %d", resp.StatusCode)
+	}
+	var target map[string][]DNSServer
+	if dErr := c.decodeJSON(resp, &target); dErr != nil {
+		return nil, fmt.Errorf("Could not decode JSON response: %v", dErr)
+	}
+	return &target["test"][0], nil
+}
+
+//DeleteDNSServer - delete dns server test
+func (c *Client) DeleteDNSServer(id int) error {
+	resp, err := c.post(fmt.Sprintf("/tests/dns-server/%d/delete", id), nil, nil)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != 204 {
+		return fmt.Errorf("failed to delete page load, response code %d", resp.StatusCode)
```

Comment:
sorry to be a pita, last thing. Then lets merge this. just need to update log msg

Reply:
No worries man, thats what code reviews are for